### PR TITLE
fix(www): fix scroll when filtering showcase

### PR DESCRIPTION
**Pull request by wardpeet**
## Description
Fixes scrolling when clicking on filters in the showcase. All filters do exactly the same now. I moved state up.

## Related Issues
#11001 
#12003

### CODE REVIEW

**Comment by wardpeet** [comment]
So I use replaceState instead of pushState for now but we probably want to fix this as well.

I can probably disable scrolling with `shouldUpdateScroll` api or use native @reach/router's navigate unsure which one is the best for this one.

I did a lot of cleanup as it was hard to follow how everything worked and not everything worked.. 😋

**Comment by t2ca** [comment]
It's working great! Just wondering if it's possible to have the url updated with the categories selected?

**Comment by t2ca** [comment]
Im not sure if its just me, but after i click "Reset all filters", i can't scroll down.

**Comment by wardpeet** [comment]
good catch thank you! I'll fix that right away, it seems to be in an infinite loop.
